graphs/adapter-jgrapht/src/main/kotlin/com/anaplan/engineering/azuki/graphs/adapter/jgrapht/check/JGraphTCheckFactory.kt

```diff
@@ -15,13 +15,14 @@ class JGraphTCheckFactory : GraphCheckFactory {
     override fun hasCycles(graphName: String, hasCycles: Boolean) =
         HasCyclesCheck(graphName, hasCycles)
 
-    override fun getSimpleCycleCount(graphName: String, count: Long) =
-        GetSimpleCycleCountCheck(graphName, count)
+    override fun hasSimpleCycleCount(graphName: String, count: Long) =
+        HasSimpleCycleCountCheck(graphName, count)
 
-    override fun hasPath(graphName: String, from: Any, to: Any, result: Boolean) =
-        HasPathCheck(graphName, from, to, result)
+    override fun pathExists(graphName: String, from: Any, to: Any) =
+        PathExistsCheck(graphName, from, to, true)
 
-}
+    override fun noPathExists(graphName: String, from: Any, to: Any) =
+        PathExistsCheck(graphName, from, to, false)
```

Comment:
🚫 I would prefer to have a single API function for this. The split should be only at the DSL level.